### PR TITLE
docs: Add "previous versions" page

### PR DESCRIPTION
**Pull request by glasser**
This links to migration guides and provides the current release phase status on each version.

### CODE REVIEW

**Comment by glasser** [comment]
@rkoron007 I put this together with @trevor-scheer and @maryelizbeth today. Thoughts?

It's "other versions" rather than "previous versions" because we'll backport it to the v3 and v2 doc sites (and I don't see why each doc site shouldn't describe the existence of all three versions).

Can you think of other parts of the v4 doc that should link to this page?  The front page? The migration guide? Etc?  (When we backport to v2 we will add something extra to the subscriptions/uploads/playground pages.)

We'll also update the README of v4 to link to it (and maybe the old readmes too, though not really worth an npm publish just for README) and then also npm deprecate the old packages.

File: docs/source/other-versions.mdx [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+Additionally, three features of Apollo Server 2 have an earlier end-of-life date: **December 31, 2022**. These features are [subscriptions](/apollo-server/v2/data/subscriptions/), [file uploads](/apollo-server/v2/data/file-uploads/), and [GraphQL Playground](/apollo-server/v2/testing/graphql-playground/). These features are based on third-party software that is unmaintained (or in the case of file uploads, based on an older version that cannot be upgraded for compatibility reasons), and are no longer part of the default experience in Apollo Server 3 or 4. Apollo does not promise that security issues or major regressions to these features will be addressed after 2022. In other words, issues that cannot be reproduced when a server is initialized with `new ApolloServer({ subscriptions: false, uploads: false, playground: false })` may not be addressed after 2022.
```
Comment (rkoron007) [inline docs/source/other-versions.mdx:25]: I think a list helps break up this content and makes it all flow a bit better. I also tweaked the bottom paragraph for style:
```suggestion
Three features of Apollo Server 2 have an earlier end-of-life date: **December 31st, 2022**. These three features are:
* [Subscriptions](/apollo-server/v2/data/subscriptions/)
* [File uploads](/apollo-server/v2/data/file-uploads/) 
* [GraphQL Playground](/apollo-server/v2/testing/graphql-playground/). 

These features incorporate third-party software that is unmaintained (or can't be upgraded for compatibility reasons). Additionally, these features aren't in the default experiences of either Apollo Server 3 or 4. Apollo generally won't support these features' security issues or significant regressions after 2022. In other words, problems that are solved when a server is initialized with `new ApolloServer({ subscriptions: false, uploads: false, playground: false })` might not be addressed after 2022.
```

**Review by rkoron007 — COMMENTED** [review]
I had a few suggestions for style but overall this looks great!

**Comment by rkoron007** [comment]
> It's "other versions" rather than "previous versions" because we'll backport it to the v3 and v2 doc sites (and I don't see why each doc site shouldn't describe the existence of all three versions).

Thinking about this further, I think it might make more sense to use the [new custom banner](https://github.com/apollographql/docs/pull/272/files)  (with a link to this article) on the V2 and V3 versions of the doc site _instead_ of having this article in each version.

I'd also like to add a note to the top of each of the following articles with a message like:
```
As of 15 November 2022, Apollo Server 2 is officially deprecated, with end-of-life scheduled for 22 October 2023.

[Learn more about deprecation and end-of-life.](/apollo-server/other-versions)
```

Articles to add notes to:
1. [AS2 articles](https://www.apollographql.com/docs/apollo-server/v2):
   - Introduction
   - Get Started
2. [AS3 articles](https://www.apollographql.com/docs/apollo-server/v3):
   - Introduction
   - Get Started
   - Choosing which package to use
   - Migrating to Apollo Server 3

Additionally, adding a custom note like this on the file uploads, subscriptions, and playground pages which goes something like this:
```
Apollo Server 2's subscriptions feature has an end-of-life date of 31st December 2022.

[Learn more about this deprecation and end-of-life.](/apollo-server/other-versions)
```

If this all sounds good (or is confusing, and you just want to see what it'd look like), I can make a PR to do all this!

File: docs/source/other-versions.mdx [hunk 2]
```diff
@@ -0,0 +1,30 @@
+---
+title: Other versions of Apollo Server
+---
+
+## Apollo Server 4 is generally available
+
+Apollo Server 4 is the latest version of Apollo Server and is [generally available](/resources/product-launch-stages#general-availability). Apollo considers Apollo Server 4 to be ready for use in a production environment and provides official support. There is currently no planned end-of-life date.
+
+## Previous deprecated versions
+
```
Comment (rkoron007) [inline docs/source/other-versions.mdx:10]: Could we also link to the docs page for discontinuing support somewhere in this article?

I feel like a note right here could feel good:

```suggestion

> [Learn more about deprecation and end-of-life.](https://www.apollographql.com/docs/resources/product-launch-stages#stages-for-discontinuing-support)

```

File: docs/source/other-versions.mdx [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+Both previous major versions of Apollo Server are currently [deprecated](/resources/product-launch-stages#deprecated) with end-of-life scheduled for **October 22, 2023**. Additionally, some features of Apollo Server v2 have an earlier end-of-life date of **December 31st, 2022**.
+
+Deprecated versions continue to receive security patches and updates to address major regressions until their end-of-life date. They typically do not receive new features and other kinds of bug fixes may or may not be backported to deprecated versions at Apollo's discretion.
```
[the remark this answers is not in the document]
Comment (glasser) [reply docs/source/other-versions.mdx:15]: That's the word we use in the actual policy, so I felt like I should use the same word?

**Comment by glasser** [comment]
I think I agree with your suggestion that the existence of the customizable banner on v2 and v3 (linking to this new page) means we don't need to "backport" the new page itself (if I'm understanding correctly).

Should we add any links to this page to any part of the AS4 site itself?

**Comment by rkoron007** [comment]
I think the AS4 migration guide would be the only place we need to plug this information. I think this would fit nicely in the very top note on the page!

....Also, if we aren't "backporting"  this page, how would you feel about renaming the article "Previous versions"? 👀

File: docs/source/other-versions.mdx (hunk 3 above)
[the remark this answers is not in the document]
Comment (rkoron007) [reply docs/source/other-versions.mdx:15]: Oh, let's use "major" then! Thanks for that context

**Review by rkoron007 — APPROVED** [review]
Thank you for adding this! ✨

**Pull request merged**